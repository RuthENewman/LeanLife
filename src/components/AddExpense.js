import React from 'react';

const AddExpense = () => (
  <div>
    Add an expense
  </div>
);

export default AddExpense;
