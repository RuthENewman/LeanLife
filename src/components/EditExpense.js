import React from 'react';
import { connect } from 'react-redux';

const EditExpense = (props) => {
  console.log(props);
  return (
    <div>
      Edit the expense with id of {props.match.params.id}.
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    expense: state.expenses.find((expense) =>  expense.id === props.match.params.id;
  };
};

export default connect(mapStateToProps)(EditExpense);
