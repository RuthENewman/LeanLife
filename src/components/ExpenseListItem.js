import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";
import "numeral/locales/en-gb";

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  numeral.locale("en-gb");
  return (
    <Link className="list-item" to={`/edit/${id}`}>
      <div>
        <h3>{description}</h3>
        <span> {moment(createdAt).format("dddd Do MMMM YYYY")}</span>
      </div>
      <h3>{numeral(amount / 100).format("$#,##0.00")}</h3>
    </Link>
  );
};

export default ExpenseListItem;
