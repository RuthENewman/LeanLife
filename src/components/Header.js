import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header>
    <h1>Lean Life</h1>
    <NavLink to="/dashboard" activeClassName="is-active">My expenses</NavLink>
    <NavLink to="/create" activeClassName="is-active">Add an expense</NavLink>
    <button onClick={startLogout}>Log out</button>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);

const db = {
  expenses: {}
}
