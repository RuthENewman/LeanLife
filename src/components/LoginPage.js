import React, { Component } from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Lean Life</h1>
      <p>Manage your expenses and streamline your life</p>
      <button className="button" onClick={startLogin}>
        Log in
      </button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
