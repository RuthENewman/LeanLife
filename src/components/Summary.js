import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import 'numeral/locales/en-gb';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const Summary = ({ count, total }) => {
  numeral.locale('en-gb');
  const word = count === 1 ? 'expense' : 'expenses';
  const formattedTotal = numeral(total / 100).format('$#,##0.00');
  return (
    <div>
      <h3>Viewing {count} expenses totalling {formattedTotal}.</h3>
    </div>
  )
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    count: visibleExpenses.length,
    total: selectExpensesTotal(visibleExpenses)
  }
};

export default connect(mapStateToProps)(Summary);
