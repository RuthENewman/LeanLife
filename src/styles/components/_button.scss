.button {
  background: $teal;
  border: none;
  border-radius: $medium-border-radius;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 350;
  line-height: 1;
  padding: $small-margin $medium-margin;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: $mid-blue;
  margin: 0 $small-margin 0 $small-margin;
}
