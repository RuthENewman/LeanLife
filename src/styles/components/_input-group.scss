.input-group {
  display: flex;
  flex-direction: column;
  margin-bottom: $medium-margin;

  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    margin: 0 $large-margin 0 $small-margin;
  }
}

.input-group__item {
  margin-bottom: $small-margin;

  @media (min-width: $desktop-breakpoint) {
    margin: 0 $small-margin 0 0;
  }
}
