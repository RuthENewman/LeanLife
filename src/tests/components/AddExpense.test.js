import React from 'react';
import { shallow } from 'enzyme';
import { AddExpense } from '../../components/AddExpense';
import expenses from '../fixtures/expenses';

let addExpense, history, wrapper;

beforeEach(() => {
  addExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddExpense
    addExpense={addExpense} history={history}
    />);
});

test('renders the Add Expense Page', () => {
  expect(wrapper).toMatchSnapshot();
});

test('handles submit', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(addExpense).toHaveBeenLastCalledWith(expenses[0]);
});
