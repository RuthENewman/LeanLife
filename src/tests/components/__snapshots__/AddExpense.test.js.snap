// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Add Expense Page 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Add an expense
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <ExpenseForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
